### ISSUE TITLE
[REVIEW]: Lidium: a 3D point cloud processing software for TLS LiDAR data

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#3623]
**Submitting author:** @nsviel (<a href="http://orcid.org/0000-0002-4608-5850">Nathan</a>)
**Repository:** <a href="https://github.com/nsviel/Lidium" target ="_blank">https://github.com/nsviel/Lidium</a>
**Version:** v1.0.0
**Editor:** @martinfleis
**Reviewer:** @lachlangrose, @hechth
**Archive:** Pending

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/6e5678fe608197d111a04ea96325b20e/status.svg)](https://joss.theoj.org/papers/6e5678fe608197d111a04ea96325b20e)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/6e5678fe608197d111a04ea96325b20e"><img src="https://joss.theoj.org/papers/6e5678fe608197d111a04ea96325b20e/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/6e5678fe608197d111a04ea96325b20e/status.svg)](https://joss.theoj.org/papers/6e5678fe608197d111a04ea96325b20e)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/nsviel/Lidium" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@lachlangrose & @hechth, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @martinfleis know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

## Review checklist for @lachlangrose
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/nsviel/Lidium">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [ ] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [ ] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [ ] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/nsviel/Lidium">repository url</a>?
- [ ] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [ ] **Contribution and authorship:** Has the submitting author (@nsviel) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [ ] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [ ] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [ ] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [ ] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [ ] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [ ] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [ ] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

## Review checklist for @hechth
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/nsviel/Lidium">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [ ] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [ ] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [ ] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/nsviel/Lidium">repository url</a>?
- [ ] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [ ] **Contribution and authorship:** Has the submitting author (@nsviel) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [ ] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [ ] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [ ] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [ ] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [ ] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [ ] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [ ] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @lachlangrose, @hechth it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
Wordcount for `paper.md` is 754

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1016/j.isprsjprs.2020.11.015 is OK
- 10.3390/s151128099 is OK
- 10.1109/ICRA.2011.5980567 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03623/joss.03623/10.21105.joss.03623.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03623/joss.03623/10.21105.joss.03623.pdf) :page_facing_up: :point_left:

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=1.65 s (289.6 files/s, 126202.9 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
C/C++ Header                   386          19139          28944         105824
C++                             76           6595           7002          37393
make                             1            683            253           1342
CMake                            9             60             35            460
JSON                             1              0              0            357
Markdown                         2             25              0             66
TeX                              1              7              0             58
F#                               1              2              0              7
Bourne Shell                     1              6              9              6
-------------------------------------------------------------------------------
SUM:                           478          26517          36243         145513
-------------------------------------------------------------------------------


Statistical information for the repository '1917f24186e12d93e75766aa' was
gathered on 2021/08/17.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
nsviel                           4        104855           2043          100.00

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
nsviel                   204708          195.2          0.0               18.10
```

**Comment by martinfleis** [comment]
👋  @nsviel, @lachlangrose, @hechth this is the review thread for the paper. All of our communications will happen here from now on.

Both reviewers have checklists at the top of this thread with the JOSS requirements. As you go over the submission, please check any items that you feel have been satisfied. There are also links to the JOSS reviewer guidelines.

The JOSS review is different from most other journals. Our goal is to work with the authors to help them meet our criteria instead of merely passing judgment on the submission. As such, the reviewers are encouraged to submit issues and pull requests on the software repository. When doing so, please mention `openjournals/joss-reviews#3623` so that a link is created to this thread (and I can keep an eye on what is happening). Please also feel free to comment and ask questions on this thread. In my experience, it is better to post comments/questions/suggestions as you come across them instead of waiting until you've reviewed the entire package.

We aim for reviews to be completed within about 2-4 weeks. Please let me know if any of you require some more time. We can also use Whedon (our bot) to set automatic reminders if you know you'll be away for a known period of time.

Please feel free to ping me (@martinfleis) if you have any questions/concerns.

**Comment by whedon** [comment]
:wave: @hechth, please update us on how your review is going (this is an automated reminder).

**Comment by whedon** [comment]
:wave: @lachlangrose, please update us on how your review is going (this is an automated reminder).

**Comment by martinfleis** [comment]
Hi @nsviel, I see that the reviewers opened issues in the Lidium repository that need to be resolved before we can move on. Do you have any guess when you will be able to work on them?

**Comment by nsviel** [comment]
Hi @martinfleis, unfortunately I don't have enough free time at the moment (as I start a new job). But I will try to resolved the opened questions as soon as possible. Some issues need a certain amount of work (automated tests, documentation), I could hope to finish at the end of September.

**Comment by hechth** [comment]
Just make sure to also keep things publicly updated so we can track the progress :)

If you have some questions or need some hints, feel free to reach out!

**Comment by martinfleis** [comment]
@nsviel Do you have any update on the progress? We don't have any deadline for the response but also don't want issues to go stale.

**Comment by martinfleis** [comment]
Hi @nsviel,

Since I wasn't able to reach you since September and there was no activity in the original repository where the reviewers left their comments, I will be forced to formally reject this submission on January 17, 2022 (two weeks from now), if we won't hear from you by then. I am really sorry to make this announcement but it is not feasible to keep the review open for such a long time without any action being done.

Thank you for understanding.

**Comment by martinfleis** [comment]
Due to the lack of activity from the authors, I recommend rejecting this paper.

@hechth, @lachlangrose Thank you for the time you spent reviewing this submission. I am sorry we are unable to finish the process towards acceptance and publishing this time.

cc @openjournals/joss-eics

**Comment by danielskatz** [comment]
@whedon reject

**Comment by whedon** [comment]
Paper rejected.
